kse/src/main/java/org/kse/gui/actions/ExportCsvAction.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2004 - 2013 Wayne Grant
+ *           2013 - 2025 Kai Kramer
+ *
+ * This file is part of KeyStore Explorer.
+ *
+ * KeyStore Explorer is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * KeyStore Explorer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with KeyStore Explorer.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.kse.gui.actions;
+
+import java.awt.Toolkit;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import javax.swing.ImageIcon;
+import javax.swing.JOptionPane;
+import javax.swing.JTable;
+import javax.swing.KeyStroke;
+import javax.swing.table.TableColumnModel;
+
+import org.kse.gui.KseFrame;
+import org.kse.gui.dialogs.DExportCsv;
+import org.kse.gui.error.DError;
+import org.kse.utilities.history.KeyStoreHistory;
+
+/**
+ * Action to export the active KeyStore table as CSV.
+ */
+public class ExportCsvAction extends KeyStoreExplorerAction {
+    private static final long serialVersionUID = 1L;
+    private static final DateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
```

Comment:
Any particular reason for adding milliseconds to the format string? For certificate validities they are always "000" and I think this might cause issues with locales where the separator is a comma instead.